transaction/_transaction.py

```diff
@@ -138,6 +138,8 @@ def user(self):
     def user(self, v):
         if v is not None:
             self._user = text_or_warn(v)
+        else:
```

Comment:
Getting back to this, let's make setting user or description to None an error, as it was (eventually) in transaction 1.